haskell-best-practices.md

```diff
@@ -505,6 +505,11 @@ The fact is that `mono-traversable` is not an abstraction that targets our
 use-case, it decreases type inference, increases cryptic type errors, and
 increases boilerplate.
 
+## Strictness
+
+Add `LANGUAGE StrictData` to modules that contain only plain records that will
```

Comment:
FWIW, I find this kind of risky. It might only contain plain records _right now_, and when that changes we're expecting folks to be good about removing the pragma?

Reply:
If you introduce a data type that needs to be lazy, then you'll need to remove `StrictData` and switch to bang patterns. :man_shrugging:

Reply:
"You'll need to" as in the compiler will force you?

Reply:
Well the compiler won't force you to do anything, but your need for lazy semantics will.

Reply:
Yeah, but my point was it's risky because you'll need to remember to do so. For example, if you introduce a recursive and infinite type and forget to remove or override the pragma, you'd introduce an infinite loop right?

Reply:
Yup, which is one of the things that makes me wary of `StrictData` and `Strict`, since they change the semantics of the language.

Reply:
You can either remove `StrictData` and use `!` or keep `StrictData` and use `~`

Reply:
But we're comfortable with requiring a human to _remember_ to do that? Perhaps a different human than the one who added the pragma to begin with? And in a PR where the diff likely won't even show the module header during review?

I think I am (hence merging this), but I think it's a valuable question to ask, and was missed in the original Issue discussion.